src/front-end/src/components/Auth/Auth.test.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { act } from 'react-dom/test-utils';
+import userEvent from '@testing-library/user-event';
+import { mockWindowMatchMedia } from '../../helper-methods';
+
+import Auth from './Auth';
+
+let container = null;
+
+mockWindowMatchMedia();
+
+beforeEach(() => {
+  container = document.createElement('div');
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  // cleanup on exiting
+  unmountComponentAtNode(container);
+  container.remove();
+  container = null;
+});
+
+it('renders without crashing', () => {
+  render(
+    <Router>
+      <Auth />
+    </Router>,
+    container,
+  );
+  expect(container.textContent).toBe(
+    'Mentor TutorialSuggested MenteesAccepted MenteesLogout',
+  );
+});
+
+it('has logout button', () => {
+  render(
+    <Router>
+      <Auth />
+    </Router>,
+    container,
+  );
+  const logoutButton = document.getElementsByTagName('button')[0];
+  expect(logoutButton.textContent).toBe('Logout');
+});
+
+it('has 3 properly-labelled links', () => {
+  render(
+    <Router>
+      <Auth />
+    </Router>,
+    container,
+  );
+  const links = document.getElementsByTagName('a');
+  expect(links[0].textContent).toBe('Mentor Tutorial');
+  expect(links[1].textContent).toBe('Suggested Mentees');
+  expect(links[2].textContent).toBe('Accepted Mentees');
+});
+
+it('has 3 links and one button', () => {
+  render(
+    <Router>
+      <Auth />
+    </Router>,
+    container,
+  );
+  const links = document.getElementsByTagName('a');
+  expect(links.length).toBe(3);
+
+  const buttons = document.getElementsByTagName('button');
+  expect(buttons.length).toBe(1);
+});
```

Comment:
I'm asking out of curiosity, not for the sake of making changes: I'm wondering if we could make these two tests into one (such as 'has 3 properly-labelled links and one button'), since the test seems to be fairly similar? I haven't done any testing until now, so I'm just wondering if it would be more error prone if one test were created rather than two?

Reply:
I'm not very used to React unit testing either, but my thinking in making these two tests separate is that if one fails, the issue that will come up will be very specific: either we have the wrong number of elements of a certain kind (test 2 fails), or the labels are wrong (test 1 fails). If we only have one test, we don't know which one it is until we investigate further. For this reason, I think it's fair to keep it as two. Happy to discuss this, as always! :)